### ISSUE TITLE
Error in session_set_save_handler ($ class, TRUE);

### ISSUE THREAD

**Issue by DevDiamondCom** [bcit-ci/CodeIgniter#4984]
Some hosting with PHP 5.4 do not support function ```php session_set_save_handler ($ class, TRUE) ;```. The server falls to the 502 error.

But in the second embodiment, when used for example as follows:
```php
session_set_save_handler(
	array($class, 'open'),
	array($class, 'close'),
	array($class, 'read'),
	array($class, 'write'),
	array($class, 'destroy'),
	array($class, 'gc')
);
register_shutdown_function('session_write_close');
```
the error was not observed in any hosting and respectively supported by all versions of php.

This fragment is taken from `/system/libraries/Session/Session.php - [Line:108] CI_v3.1.3`.

I advise you to put the conditions in PHP version on 7. So for example:
```php
	if (is_php('7'))
	{
		session_set_save_handler($class, TRUE);
	}
	else
	{
		session_set_save_handler(
			array($class, 'open'),
			array($class, 'close'),
			array($class, 'read'),
			array($class, 'write'),
			array($class, 'destroy'),
			array($class, 'gc')
		);

		register_shutdown_function('session_write_close');
	}
```

**Comment by narfbg** [comment]
"Some hosing"? You'll have to be more specific than that.

**Comment by DevDiamondCom** [comment]
I beg your pardon. I wanted to say "shared hostings".
Just wrote one game server (engine), and 40% of the questions of the users are almost identical: "Install the engine and throws out an error (502)."

We conducted a study and found that some "shared hosting" when you call the php function «session_set_save_handler ()» with class transfer, arises "502 bad gateway"
And PHP7 and above did not notice it.

**Comment by narfbg** [comment]
"Some shared hosting" doesn't really give any more detail than "some hosting" ... it's only natural that I assume you're talking about sharing hosting environments. That's not the point.

The feature in question was introduced in PHP 5.4.0, so avoiding it's usage on everything below 7.0 sounds very counter-productive.

You've not said which PHP versions you've experienced this issue with, on which and how many hosting providers, nor have you given us any real debugging info ... 502 is an HTTP status code that as far as PHP goes, only says "something went wrong".
It is also unclear how you've tracked it down to that `session_set_save_handler()` call.

In summary: We need as many details possible and you're giving us the opposite.

**Comment by DevDiamondCom** [comment]
OK. Now the will give examples.

**Comment by DevDiamondCom** [comment]
That filled for example the latest CI - CodeIgniter-3.1.3

PHP Info:
[http://issue-4984.jobs.devdiamond.com/php_info.php](http://issue-4984.jobs.devdiamond.com/php_info.php)

CI
[http://issue-4984.jobs.devdiamond.com/CodeIgniter-3-1-3](http://issue-4984.jobs.devdiamond.com/CodeIgniter-3-1-3)
1. Add to autoload rule session «session»

CI
[http://issue-4984.jobs.devdiamond.com/CodeIgniter-3-1-3_mod](http://issue-4984.jobs.devdiamond.com/CodeIgniter-3-1-3_mod)
1. Add to autoload rule session «session»
2. Change the code in the ```php system/libraries/Session/Session.php``` [Line:108] with ```php if (is_php ( '5.4'))``` to ```php if (is_php ( '5.5'))```.

Accordingly, it is prescribed in the engine ```php if (is_php ( '5.5'))``` and projects for all customers working stably. PHP7 I said as the max, as the design which was introduced in ```php session_set_save_handler ()``` function to PHP5.4 (ie with parameter transfer "open", "close" etc.) works 100% to version PHP7. But in practice yet and PHP5.5 and above is stable.

As I noticed, in version CI-3, which is currently in the GitHub, generally removed checker for ```php session_set_save_handler ()``` function is used to transfer the class. But as practice shows, all hosting has not yet updated the PHP product at least 5.5.

Old stable version 5.5.38, and it is not removed.

If I'm wrong, then prescribe for all development conditions of use PHP5.5 version and above.

**Comment by narfbg** [comment]
PHP error logs?

**Comment by DevDiamondCom** [comment]
[error.log.txt](https://github.com/bcit-ci/CodeIgniter/files/709279/error.log.txt)
[access.log.txt](https://github.com/bcit-ci/CodeIgniter/files/709280/access.log.txt)

**Comment by narfbg** [comment]
PHP, not HTTP ... the latter can't tell me anything more than 502.

**Comment by DevDiamondCom** [comment]
The log does not say anything. Just cut down PHP and all. But all the same, that all the logs, after the establishment of subdomains.

**Comment by DevDiamondCom** [comment]
Well, if it's not really a question, I think I do better in the documentation indicating the minimum requirement for the written product, and all.

I just wanted to help, but on reflection, came to the conclusion that all the same, after some 6-7 months PHP5.5 also officially closed, and all the hosting companies must compulsorily pass on PHP5.5 or PHP5.6 minimum.

**Comment by narfbg** [comment]
I don't know how to reply to this.

The logs can't say nothing unless you've disabled them ... this is PHP debugging 101.

**Comment by DevDiamondCom** [comment]
Well, I'm too shocked. Including the withdrawal of all the errors and mistakes can not be recorded. Just cuts off at a meeting of the PHP functions and all. Although I think we can close the topic and, as described above. Just specify the PHP documentation minutes vesiyu all. Since still PHP5.4 dyal many are not suitable projects.
What do you say, you can close the topic?

**Comment by narfbg** [comment]
This is not a forum board ... there's no topics. It's a bug tracker, and I'm having a very hard time trying to understand anything you say.
It looks like you're using Google Translator or something, which only makes it worse.

But sure - closed.

**Comment by DevDiamondCom** [comment]
I beg your pardon. English has not yet learned well, and use Google Translate.
I know that the bug tracker, and it wrote. I wanted help, but it seems wrong. And now, just so close, I thought meanly, asked about this.
Excuse me please that it took your time :)
CodeIgniter is very cool, and easy.
